src/Compilers/CSharp/Test/Syntax/Parsing/RecordParsing.cs

```diff
@@ -3145,11 +3145,85 @@ public void RecordRecordParsing()
             EOF();
         }
 
-        [Fact, CompilerTrait(CompilerFeature.RecordStructs)]
-        public void RecordStructParsing_WrongOrder()
+        [Fact, CompilerTrait(CompilerFeature.RecordStructs), WorkItem(59534, "https://github.com/dotnet/roslyn/issues/59534")]
+        public void RecordStructParsing_WrongOrder_CSharp10()
         {
             var text = "struct record C(int X, int Y);";
-            UsingTree(text, options: TestOptions.RegularPreview,
+            UsingTree(text, options: TestOptions.Regular10,
+                // (1,8): error CS9012: Unexpected keyword 'record'. Did you mean 'struct record' or 'class record'?
+                // struct record C(int X, int Y);
+                Diagnostic(ErrorCode.ERR_MisplacedRecord, "record").WithLocation(1, 8),
+                // (1,16): error CS1514: { expected
+                // struct record C(int X, int Y);
+                Diagnostic(ErrorCode.ERR_LbraceExpected, "(").WithLocation(1, 16),
```

Comment:
this is a pity.  consider (if cheap) supporting this as a record, with misplaced `struct`.   Once you see 'record' (even if it's errant) i think we should exercise the record-parsing path.

Reply:
the primary reason for this is so that the tree doesn't go off the rails.  that helps the entire experience.

Reply:
I understand, but prefer to leave as-is.
It's not trivial because the keyword is `struct` (not `record`) and so we'd have to loosen assertions in `constructTypeDeclaration(...)` below which I'd rather not.
This will likely resolve when we allow parameter list on type declaration for primary constructors.

Reply:
That works for me!  Thanks :)